tools/base/fpgad/CMakeLists.txt

```diff
@@ -62,12 +62,18 @@ target_link_libraries(fpgad
 
 set(CMAKE_C_FLAGS "-DHAVE_CONFIG_H ${CMAKE_C_FLAGS} -pthread")
 
+configure_file(fpgad.service.in fpgad.service @ONLY NEWLINE_STYLE UNIX)
+
 install(TARGETS fpgad
         RUNTIME DESTINATION bin
         COMPONENT toolfpgad)
 
 if ("${CMAKE_INSTALL_PREFIX}" STREQUAL "/usr")
   install(FILES fpgad.cfg DESTINATION /etc/opae COMPONENT toolfpgad)
+  install(FILES fpgad.conf DESTINATION /etc/sysconfig COMPONENT toolfpgad)
 else()
   install(FILES fpgad.cfg DESTINATION ${CMAKE_INSTALL_PREFIX}/etc/opae COMPONENT toolfpgad)
+  install(FILES fpgad.conf DESTINATION ${CMAKE_INSTALL_PREFIX}/etc/sysconfig COMPONENT toolfpgad)
 endif()
+
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/fpgad.service DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/systemd/system COMPONENT toolfpgad)
```

Comment:
Have you tested this?  I was unable to get the files to install automatically.  I think the "real" path is /etc/systemd/system

Reply:
I did, yes..  there was a problem with the definition of CMAKE_INSTALL_PREFIX.  The tools directory was being traversed before the variable had been set.  Thanks to @nakulkorde for noticing the issue.  Ok, I will look into changing it to /etc/systemd/system.  Thanks!

Reply:
OK.  Just to be clear, the service files are in /lib/systemd/system for Ubuntu, but /etc/systemd/system for CentOS/RHEL.  I believe this was one sticky point I had with my implementation.

I also meant that you need to install the .rpd or .deb files on a clean system to see whether they got installed in the proper place, and the '@' substitution put in the correct paths.  You may have done that...

Reply:
Yep, I verified that the files were going to the right spot.
Re: the target directory for the .service, I checked on one of our RHEL 7.6 boxes, and it looks like /usr/lib/systemd/system is the right place.  I see several other .service files there.  Seem OK?

Reply:
Look [here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/chap-Managing_Services_with_systemd#tabl-Managing_Services_with_systemd-Introduction-Units-Locations).  On the minicloud machine DCP uses for testing (CentOS 7) it is in /etc/systemd/system.

I believe we need to 'systemctl enable' these services, which implies (from the table) that they belong in /etc.

Reply:
In thinking about this, I believe the current approach - installing in opae directory and requiring admin to edit and move them - makes the most sense.  These files will be in different places depending on the whim of the admin.  system.conf can redirect these directories anywhere one wants.

Reply:
Ok, good point.  Let me seek guidance on the one-click requirement.

Reply:
I was told the preference was to install the files in their final destinations.